src/core/thread/mesh_forwarder.cpp

```diff
@@ -802,8 +821,27 @@ ThreadError MeshForwarder::SendFragment(Message &aMessage, Mac::Frame &aFrame)
         fcf |= Mac::Frame::kFcfSecurityEnabled;
     }
 
+    if (aMessage.IsMleDiscoverRequest())
+    {
+        dstpan = Mac::kPanIdBroadcast;
```

Comment:
Is it necessary to process the case that to discover a specific dst PAN ID instead of broadcast PAN ID? not sure how joiner knows the specific dst PAN ID before it joins the thread network.